resources/[soz]/soz-core/src/nui/components/Shop/ClothShopMenu.tsx

```diff
@@ -0,0 +1,183 @@
+    MenuItemButton,
+    MenuItemCheckbox,
+    MenuItemSelect,
+    MenuItemSelectOption,
+    MenuTitle,
+    SubMenu,
+} from '../Styleguide/Menu';
+
+type MenuClothShopStateProps = {
+    catalog: {
+        brand: string;
+        shop_content: ClothingShop;
+        shop_categories: Record<number, ClothingShopCategory>;
+        player_data: PlayerData;
+        under_types: Record<number, number[]>;
+    };
+};
+
+export const ClothShopMenu: FunctionComponent<MenuClothShopStateProps> = ({ catalog }) => {
+    const banner = BrandsConfig[catalog.brand as ShopBrand].banner || 'https://nui-img/soz/menu_shop_clothe_normal';
+    const shopName = BrandsConfig[catalog.brand as ShopBrand].label || 'Magasin';
+    const navigate = useNavigate();
+    const location = useLocation();
+    const state = location.state as { activeIndex: number } | undefined;
+    const [stocks, setStocks] = useState<Record<number, number>>(catalog.shop_content.stocks);
+    const [playerData, setPlayerData] = useState<PlayerData>(catalog.player_data);
+
+    const selectCategory = (categoryId: number) => {
+        navigate(`/${MenuType.ClothShop}/${categoryId}`, {
+            state: {
+                ...(state || {}),
+                activeIndex: 0,
+            },
+        });
+    };
+
+    useNuiEvent('menu', 'Backspace', () => {
+        fetchNui(NuiEvent.ClothingShopBackspace);
+    });
+
+    useNuiEvent('cloth_shop', 'SetStocks', stocks => {
+        setStocks(stocks);
+    });
+
+    useNuiEvent('cloth_shop', 'SetPlayerData', playerData => {
+        setPlayerData(playerData);
+    });
+
+    return (
+        <Menu type={MenuType.ClothShop}>
+            <MainMenu>
+                <MenuTitle banner={banner}>{shopName}</MenuTitle>
+                <MenuContent>
+                    <MenuItemCheckbox
+                        onChange={check => {
+                            fetchNui(NuiEvent.ClothShopToggleCamera, check);
+                        }}
+                    >
+                        Libérer la caméra
+                    </MenuItemCheckbox>
+                    {Object.values(catalog.shop_content.categories)
+                        .filter(
+                            category =>
+                                (Object.values(catalog.shop_categories[category.id].content).length > 0 ||
+                                    Object.values(catalog.shop_categories).filter(
+                                        childCat => childCat.parentId == category.id
+                                    ).length > 0) &&
+                                (category.id != ClothingCategoryID.UNDERSHIRTS ||
+                                    (playerData.cloth_config.BaseClothSet.TopID != null &&
+                                        catalog.under_types[playerData.cloth_config.BaseClothSet.TopID] != null &&
+                                        catalog.under_types[playerData.cloth_config.BaseClothSet.TopID].length > 0))
+                        )
```

Comment:
Please move this filter out of JSX. It's more readable if you use a constant with a clear name

Reply:
Done